projects/addon-charts/components/pie-chart/pie-chart.directive.ts

```diff
@@ -0,0 +1,63 @@
+import {Directive, ElementRef, Inject, Input, NgZone} from '@angular/core';
+import {ANIMATION_FRAME, PERFORMANCE} from '@ng-web-apis/common';
+import {describeSector} from '@taiga-ui/addon-charts/utils';
+import {
+    clamp,
+    easeInOutQuad,
+    tuiDefaultProp,
+    TuiDestroyService,
+    tuiZonefree,
+} from '@taiga-ui/cdk';
+import {TUI_ANIMATIONS_DURATION} from '@taiga-ui/core';
+import {BehaviorSubject, Observable} from 'rxjs';
+import {map, pairwise, switchMap, takeUntil, takeWhile} from 'rxjs/operators';
+
+// @dynamic
+@Directive({
+    selector: 'path[tuiPieChart]',
+    providers: [TuiDestroyService],
+})
+export class TuiPieChartDirective {
+    private readonly sector$ = new BehaviorSubject<readonly [number, number]>([0, 0]);
+
+    @Input()
+    @tuiDefaultProp()
+    set tuiPieChart(sector: readonly [number, number]) {
+        this.sector$.next(sector);
+    }
+
+    constructor(
+        @Inject(ElementRef) {nativeElement}: ElementRef<SVGPathElement>,
+        @Inject(NgZone) ngZone: NgZone,
+        @Inject(TuiDestroyService) destroy$: Observable<unknown>,
+        @Inject(PERFORMANCE) performance: Performance,
+        @Inject(ANIMATION_FRAME) animationFrame$: Observable<number>,
+        @Inject(TUI_ANIMATIONS_DURATION) duration: number,
+    ) {
+        this.sector$
+            .pipe(
+                pairwise(),
+                switchMap(([prev, cur]) => {
```

Comment:
can we move the code of the `switchMap body` function and test it separately?

Reply:
We would have to store `animationFrame$` and `performance` as class properties and to test it we would need to mock a lot of stuff just to test simple math + quadruple easing function would make it hard to match actual values.